.github/workflows/changelog-check.yml

```diff
@@ -38,6 +38,9 @@ jobs:
             # "Enhancement is not a valid type of changelog entry, but it's a common mistake.
             echo "::error::Found invalid type (enhancement) in changelog - did you mean improvement?"
             exit 1
+          elif grep -q ':deprecate$' $changelog_files; then
```

Comment:
Seems like we should have a final case that checks for all known valid strings and errors if it isn't one of them.